sway-lib-std/src/context/call_frames.sw

```diff
@@ -44,18 +50,12 @@ pub fn code_size() -> u64 {
 
 /// Get the first parameter from the current call frame.
 pub fn first_param() -> u64 {
-    asm(size, ptr, offset: 584) {
-        add size fp offset;
-        size: u64
-    }
+    read(frame_ptr() + FIRST_PARAMETER_OFFSET)
 }
 
 /// Get the second parameter from the current call frame.
-pub fn second_param() -> u64 {
-    asm(size, ptr, offset: 592) {
-        add size fp offset;
-        size: u64
-    }
+pub fn second_param<T>() -> T {
```

Comment:
I also think the second parameter doesn't need to be a generic. The second parameter itself is a u64 and in your tests T is actually a u64 that you then treat as a pointer and read from. I think if you change T to u64, everything should continue to work as expected.